spark/src/main/scala/geotrellis/spark/io/hadoop/HadoopGeoTiffRDD.scala

```diff
@@ -87,22 +92,54 @@ object HadoopGeoTiffRDD {
     * @param uriToKey function to transform input key basing on the URI information.
     * @param options  An instance of [[Options]] that contains any user defined or default settings.
     */
-  def apply[I, K, V](path: Path, uriToKey: (URI, I) => K, options: Options)(implicit sc: SparkContext, rr: RasterReader[Options, (I, V)]): RDD[(K, V)] = {
-    val conf = configuration(path, options)
-    options.maxTileSize match {
-      case Some(tileSize) =>
+  def apply[I, K, V](
+    path: Path,
+    uriToKey: (URI, I) => K,
+    options: Options
+  )(implicit sc: SparkContext, rr: RasterReader[Options, (I, V)]): RDD[(K, V)] = {
+
+    val conf = new SerializableConfiguration(configuration(path, options))
+    val path2 = path.toString
```

Comment:
Well it's always a bit confusing when you see in code such names. In fact you can write just 

```scala
HadoopGeoTiffInfoReader(path.toString, conf, options.tiffExtensions)
```